Within subject updates

steveharoz opened the pull request:
1. Better explained the sources of noise in the simulation
2. Show a graph of the data before analyzing
3. Parallel bootstrapping (only 100 iterations for sanity reasons)
4. Show confidence interval of effect sizes.

The bootstrapping especially needs a clear explanation.

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
````diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
     color = factor(color)
   )
 ```
 
````
Half of this section concerns data generation. Would it make the exemplar more accessible if we move the data generation to an appendix?

steveharoz replied:
It's somewhat important that people understand how the per-subject noise impacts the simulations. But I've pulled all of the data generation into one large section.

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
````diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+anovaResults %>% as.tibble()
+```
+
+Looking at the `F` and `ges` (generalized eta squared) columns, there appear to be strong main effects for `layout` and `size` and an interaction between `layout` and `size`. However `color` and the other 2-way and 3-way interactions show only negligeable effects.
````
Could you elaborate what you meant by "strong"? Did you interpret `F` and `ges` relative to other effects? Or did you interpret them relative to any criteria?

steveharoz replied:
I revised it to be relative

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+
+<mark> Draft. Needs work. </mark>
+
+But that only gives us one value, whereas we want a confidence interval.
```
I believe you meant "one point estimate per effect", right?

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+1. Do that many times
+1. Sort the results for each effect and find the 95% confidence interval
+
+
```
"Bootstrapping works because of the [central limit theorem](https://en.wikipedia.org/wiki/Central_limit_theorem). This allows us to get calculate an interval estimate (95% confidence interval of `ges`) rather than a point estimate (`ges`)

steveharoz replied:
I'd rather not rely on a reader's understanding of the CLM. I remember the concept of bootstrapping taking me a while to grasp in grad school. Here's a first pass at a simple explanation of bootstrapping:

We'll use a technique called bootstrapping, which checks the effect size for many randomized samples of the data. Importantly, bootstrapping samples "with replacement", meaning that items can be sampled more than once or not at all. If a small subset of the observations are driving an effect, they won't impact all of the samples. Consequently, the spread of the bootstrapped confidence intervals shows how consistent the results are for different samples.

dragice replied:
Here's a short explanation I used: "Briefly, it consists of generating many alternative datasets from the experimental data by randomly drawing observations with replacement. The variability across these datasets is assumed to approximate sampling error and is used to compute so- called bootstrap confidence intervals."

dragice replied:
I think the key is that (in most cases) the distribution of bootstrap samples approximates the sampling distribution.

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+  bootSubjects <- sample(subjects, length(subjects), replace=TRUE)
+  
+  # get all the results for one subject
+  # and give them a new uniwue subject id
```
"unique"

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
````diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+```{r within-bootstrap-by-subject}
+
+# run one iteration of the bootstrap procedure
+analyzeIteration <- function(x) {
````
Could you use snake case for the names of variable and functions? 

This is specified in [tidyverse style guide](http://style.tidyverse.org/syntax.html#object-names), which was referred to by [our style guide](https://github.com/transparentstats/guidelines/wiki/Style-Guide).

steveharoz replied:
Good catch. I'm going to update all naming stuff in a separate pass, to worry about text/structure first.

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+  bootResults %>% 
+  group_by(effect) %>% 
+  summarize(
+    ESPercentileLo = unname(quantile(ges, probs = PERCENTILE_LO)),
```
How about naming these columns `ges_conf_low` and `ges_conf_high`?

steveharoz replied:
(see above)

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+dataAggregated <- data %>% 
+  group_by(layout, size, color, subject) %>% 
+  summarize(response_time = mean(response_time))
+
```
Would it make sense to move the definition of dataAggregated to before the line 175? This will allow readers to read from top-to-bottom.

chatchavan commented on guides/effectsize_exemplar_within.Rmd:
```diff
@@ -31,93 +31,266 @@ format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdi
+  scale_x_continuous(expand = c(0,0)) + expand_limits(x = -0.05) +
+  expand_limits(x = max(anovaResults$ESPercentileHi) * 1.05) +
+  labs(x=expression(paste('Effect size ', eta[G]^2)), y='Effect') +
+  theme_bw()
```
Here's a suggestion to simplify and separate concerns (manipulation from plotting):

```r
import::from(forcats, fct_rev, fct_inorder)

p_anova_results <- 
anovaResults %>% 
  
  # reverse order the factors so that they appear in proper order in the plot
  mutate(effect = fct_rev(fct_inorder(effect))) %>% 
  
  # plot and mapping
  ggplot(aes(x = effect, y = ges, ymin = ESPercentileLo, ymax = ESPercentileHi)) +
  
    # reference line of no-effect
    geom_hline(yintercept = 0, linetype = 'dotted') +
    
    # point- and interval-estimates
    geom_pointrange() +
  
    # ensures that we see the reference line
    expand_limits(x = 0) +
    
    # flip to horizontal plot and apply black-and-white theme
    coord_flip() +
    theme_bw()
```

steveharoz commented:
I think I addressed all of the comments. The text still needs work, but it may be something the SIG participants can iterate on. 
@chatchavan: Do you know why travis is failing? The book builds on my machine.

chatchavan commented:
Travis fails because of some package conflict. In the last version, I built it manually (and get a versioned DOI) before putting it in the gh_pages branch. The process is quite puzzling, so I can work on that later today.




> On 23. Apr 2018, at 10:26, Steve Haroz <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> I think I addressed all of the comments. The text still needs work, but it may be something the SIG participants can iterate on.
> @chatchavan <https://github.com/chatchavan>: Do you know why travis is failing? The book builds on my machine.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/transparentstats/guidelines/pull/92#issuecomment-383594588>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AFUDaZXR-mW6WkEue71cHjx1s-XAXypBks5treSfgaJpZM4TfPAz>.
>
